### PR TITLE
Scheduler improvement

### PR DESCRIPTION
**Pull request by chengshifan**
1. use requests core max(0.3, limit/2) for job pod
2. always select the cluster which has more cpu/memory capacities.

### CODE REVIEW

File: src/api/handlers/job_api.py [hunk 1]
```diff
@@ -704,12 +704,21 @@ def assign_cluster(self, jobs):
...
+                    r = g.db.execute_one_dict('''
+                        SELECT cpu_capacity, memory_capacity FROM cluster WHERE name = %s
+                    ''', [c['name']])
+                    if r['cpu_capacity'] > max_cpu_capacity and r['memory_capacity'] > max_memory_capacity:
+                        target_cluster = c['name']
+                        max_cpu_capacity = r['cpu_capacity']
+                        max_memory_capacity = r['memory_capacity']
                 # use the parent cluster
-                for d in j.get('depends_on', []):
-                    target_cluster = assigned_clusters.get(d['job'], None)
```
Comment (iberryful) [inline src/api/handlers/job_api.py:711]: Keeping children job within the same cluster is important to performance.

File: src/scheduler/kubernetes/scheduler.py [hunk 2]
```diff
@@ -677,6 +677,10 @@ def kube_job(self, job_id, cpu, mem, services=None):
                             'limits': {
                                 'memory': '%sMi' % mem,
                                 'cpu': cpu
+                            },
+                            'requests': {
+                                'memory': '%sMi' % mem,
+                                'cpu': max(0.3, cpu / 2)
```
Comment (iberryful) [inline src/scheduler/kubernetes/scheduler.py:683]: cpu / 2.0

**Review by iberryful — CHANGES_REQUESTED** [review]

File: src/api/handlers/job_api.py (hunk 1 above)
Comment (chengshifan) [reply src/api/handlers/job_api.py:711]: OK. I will keep it available

File: src/scheduler/kubernetes/scheduler.py (hunk 2 above)
Comment (chengshifan) [reply src/scheduler/kubernetes/scheduler.py]: done

**Pull request merged**